core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2072,6 +2084,13 @@ public void onResponse(final Releasable releasable) {
                                     primaryTerm);
                             onPermitAcquired.onFailure(new IllegalStateException(message));
                         } else {
+                            try {
+                                updateGlobalCheckpointOnReplica(globalCheckpoint);
```

Comment:
I found it easier to just do the "double"-update than introducing a boolean. Do you have a preference?